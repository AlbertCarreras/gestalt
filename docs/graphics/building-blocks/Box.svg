<svg width="325" height="180" viewBox="0 0 325 180" fill="none" xmlns="http://www.w3.org/2000/svg">
<rect width="325" height="180" fill="#F4F4F4"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M49 40C49 39.4477 49.4477 39 50 39H54.1667C54.7189 39 55.1667 39.4477 55.1667 40C55.1667 40.5523 54.7189 41 54.1667 41H51V44.1667C51 44.719 50.5523 45.1667 50 45.1667C49.4477 45.1667 49 44.719 49 44.1667V40ZM61.5 40C61.5 39.4477 61.9477 39 62.5 39H70.8333C71.3856 39 71.8333 39.4477 71.8333 40C71.8333 40.5523 71.3856 41 70.8333 41H62.5C61.9477 41 61.5 40.5523 61.5 40ZM78.1667 40C78.1667 39.4477 78.6144 39 79.1667 39H87.5C88.0523 39 88.5 39.4477 88.5 40C88.5 40.5523 88.0523 41 87.5 41H79.1667C78.6144 41 78.1667 40.5523 78.1667 40ZM94.8333 40C94.8333 39.4477 95.2811 39 95.8333 39H104.167C104.719 39 105.167 39.4477 105.167 40C105.167 40.5523 104.719 41 104.167 41H95.8333C95.2811 41 94.8333 40.5523 94.8333 40ZM111.5 40C111.5 39.4477 111.948 39 112.5 39H120.833C121.386 39 121.833 39.4477 121.833 40C121.833 40.5523 121.386 41 120.833 41H112.5C111.948 41 111.5 40.5523 111.5 40ZM128.167 40C128.167 39.4477 128.614 39 129.167 39H137.5C138.052 39 138.5 39.4477 138.5 40C138.5 40.5523 138.052 41 137.5 41H129.167C128.614 41 128.167 40.5523 128.167 40ZM144.833 40C144.833 39.4477 145.281 39 145.833 39H150C150.552 39 151 39.4477 151 40V44.1667C151 44.7189 150.552 45.1667 150 45.1667C149.448 45.1667 149 44.7189 149 44.1667V41H145.833C145.281 41 144.833 40.5523 144.833 40ZM50 51.5C50.5523 51.5 51 51.9477 51 52.5V60.8333C51 61.3856 50.5523 61.8333 50 61.8333C49.4477 61.8333 49 61.3856 49 60.8333V52.5C49 51.9477 49.4477 51.5 50 51.5ZM150 51.5C150.552 51.5 151 51.9477 151 52.5V60.8333C151 61.3856 150.552 61.8333 150 61.8333C149.448 61.8333 149 61.3856 149 60.8333V52.5C149 51.9477 149.448 51.5 150 51.5ZM50 68.1667C50.5523 68.1667 51 68.6144 51 69.1667V77.5C51 78.0523 50.5523 78.5 50 78.5C49.4477 78.5 49 78.0523 49 77.5V69.1667C49 68.6144 49.4477 68.1667 50 68.1667ZM150 68.1667C150.552 68.1667 151 68.6144 151 69.1667V77.5C151 78.0523 150.552 78.5 150 78.5C149.448 78.5 149 78.0523 149 77.5V69.1667C149 68.6144 149.448 68.1667 150 68.1667ZM50 84.8333C50.5523 84.8333 51 85.281 51 85.8333V94.1667C51 94.7189 50.5523 95.1667 50 95.1667C49.4477 95.1667 49 94.7189 49 94.1667V85.8333C49 85.281 49.4477 84.8333 50 84.8333ZM150 84.8333C150.552 84.8333 151 85.2811 151 85.8333V94.1667C151 94.719 150.552 95.1667 150 95.1667C149.448 95.1667 149 94.719 149 94.1667V85.8333C149 85.2811 149.448 84.8333 150 84.8333ZM50 101.5C50.5523 101.5 51 101.948 51 102.5V110.833C51 111.386 50.5523 111.833 50 111.833C49.4477 111.833 49 111.386 49 110.833V102.5C49 101.948 49.4477 101.5 50 101.5ZM150 101.5C150.552 101.5 151 101.948 151 102.5V110.833C151 111.386 150.552 111.833 150 111.833C149.448 111.833 149 111.386 149 110.833V102.5C149 101.948 149.448 101.5 150 101.5ZM50 118.167C50.5523 118.167 51 118.614 51 119.167V127.5C51 128.052 50.5523 128.5 50 128.5C49.4477 128.5 49 128.052 49 127.5V119.167C49 118.614 49.4477 118.167 50 118.167ZM150 118.167C150.552 118.167 151 118.614 151 119.167V127.5C151 128.052 150.552 128.5 150 128.5C149.448 128.5 149 128.052 149 127.5V119.167C149 118.614 149.448 118.167 150 118.167ZM50 134.833C50.5523 134.833 51 135.281 51 135.833V139H54.1667C54.719 139 55.1667 139.448 55.1667 140C55.1667 140.552 54.719 141 54.1667 141H50C49.4477 141 49 140.552 49 140V135.833C49 135.281 49.4477 134.833 50 134.833ZM150 134.833C150.552 134.833 151 135.281 151 135.833V140C151 140.552 150.552 141 150 141H145.833C145.281 141 144.833 140.552 144.833 140C144.833 139.448 145.281 139 145.833 139H149V135.833C149 135.281 149.448 134.833 150 134.833ZM61.5 140C61.5 139.448 61.9477 139 62.5 139H70.8333C71.3856 139 71.8333 139.448 71.8333 140C71.8333 140.552 71.3856 141 70.8333 141H62.5C61.9477 141 61.5 140.552 61.5 140ZM78.1667 140C78.1667 139.448 78.6144 139 79.1667 139H87.5C88.0523 139 88.5 139.448 88.5 140C88.5 140.552 88.0523 141 87.5 141H79.1667C78.6144 141 78.1667 140.552 78.1667 140ZM94.8333 140C94.8333 139.448 95.281 139 95.8333 139H104.167C104.719 139 105.167 139.448 105.167 140C105.167 140.552 104.719 141 104.167 141H95.8333C95.281 141 94.8333 140.552 94.8333 140ZM111.5 140C111.5 139.448 111.948 139 112.5 139H120.833C121.386 139 121.833 139.448 121.833 140C121.833 140.552 121.386 141 120.833 141H112.5C111.948 141 111.5 140.552 111.5 140ZM128.167 140C128.167 139.448 128.614 139 129.167 139H137.5C138.052 139 138.5 139.448 138.5 140C138.5 140.552 138.052 141 137.5 141H129.167C128.614 141 128.167 140.552 128.167 140Z" fill="#4D4D4D"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M190 41C188.997 41 188.018 41.0983 187.071 41.2855C186.53 41.3927 186.004 41.0404 185.896 40.4986C185.789 39.9568 186.142 39.4307 186.683 39.3236C187.757 39.1112 188.866 39 190 39H194.25C194.802 39 195.25 39.4477 195.25 40C195.25 40.5523 194.802 41 194.25 41H190ZM201.75 40C201.75 39.4477 202.198 39 202.75 39H211.25C211.802 39 212.25 39.4477 212.25 40C212.25 40.5523 211.802 41 211.25 41H202.75C202.198 41 201.75 40.5523 201.75 40ZM218.75 40C218.75 39.4477 219.198 39 219.75 39H228.25C228.802 39 229.25 39.4477 229.25 40C229.25 40.5523 228.802 41 228.25 41H219.75C219.198 41 218.75 40.5523 218.75 40ZM235.75 40C235.75 39.4477 236.198 39 236.75 39H245.25C245.802 39 246.25 39.4477 246.25 40C246.25 40.5523 245.802 41 245.25 41H236.75C236.198 41 235.75 40.5523 235.75 40ZM252.75 40C252.75 39.4477 253.198 39 253.75 39H258C259.134 39 260.243 39.1112 261.317 39.3236C261.858 39.4307 262.211 39.9568 262.104 40.4986C261.996 41.0404 261.47 41.3927 260.929 41.2855C259.982 41.0983 259.003 41 258 41H253.75C253.198 41 252.75 40.5523 252.75 40ZM181.941 42.1386C182.248 42.5976 182.125 43.2187 181.666 43.5259C180.03 44.6213 178.621 46.0303 177.526 47.6665C177.219 48.1254 176.598 48.2484 176.139 47.9411C175.68 47.6339 175.557 47.0128 175.864 46.5538C177.105 44.7004 178.7 43.1048 180.554 41.864C181.013 41.5567 181.634 41.6797 181.941 42.1386ZM266.059 42.1386C266.366 41.6797 266.987 41.5567 267.446 41.864C269.3 43.1048 270.895 44.7004 272.136 46.5538C272.443 47.0128 272.32 47.6339 271.861 47.9411C271.402 48.2484 270.781 48.1254 270.474 47.6665C269.379 46.0303 267.97 44.6213 266.334 43.5259C265.875 43.2187 265.752 42.5976 266.059 42.1386ZM273.501 51.8964C274.043 51.7892 274.569 52.1416 274.676 52.6833C274.889 53.7568 275 54.8659 275 56V60.25C275 60.8023 274.552 61.25 274 61.25C273.448 61.25 273 60.8023 273 60.25V56C273 54.9969 272.902 54.0178 272.714 53.0715C272.607 52.5297 272.96 52.0036 273.501 51.8964ZM174.499 51.8964C175.04 52.0036 175.393 52.5297 175.286 53.0715C175.098 54.0178 175 54.9969 175 56V60.25C175 60.8023 174.552 61.25 174 61.25C173.448 61.25 173 60.8023 173 60.25V56C173 54.8659 173.111 53.7568 173.324 52.6833C173.431 52.1416 173.957 51.7892 174.499 51.8964ZM274 67.75C274.552 67.75 275 68.1977 275 68.75V77.25C275 77.8023 274.552 78.25 274 78.25C273.448 78.25 273 77.8023 273 77.25V68.75C273 68.1977 273.448 67.75 274 67.75ZM174 67.75C174.552 67.75 175 68.1977 175 68.75V77.25C175 77.8023 174.552 78.25 174 78.25C173.448 78.25 173 77.8023 173 77.25V68.75C173 68.1977 173.448 67.75 174 67.75ZM174 84.75C174.552 84.75 175 85.1977 175 85.75V94.25C175 94.8023 174.552 95.25 174 95.25C173.448 95.25 173 94.8023 173 94.25V85.75C173 85.1977 173.448 84.75 174 84.75ZM274 84.75C274.552 84.75 275 85.1977 275 85.75V94.25C275 94.8023 274.552 95.25 274 95.25C273.448 95.25 273 94.8023 273 94.25V85.75C273 85.1977 273.448 84.75 274 84.75ZM274 101.75C274.552 101.75 275 102.198 275 102.75V111.25C275 111.802 274.552 112.25 274 112.25C273.448 112.25 273 111.802 273 111.25V102.75C273 102.198 273.448 101.75 274 101.75ZM174 101.75C174.552 101.75 175 102.198 175 102.75V111.25C175 111.802 174.552 112.25 174 112.25C173.448 112.25 173 111.802 173 111.25V102.75C173 102.198 173.448 101.75 174 101.75ZM174 118.75C174.552 118.75 175 119.198 175 119.75V124C175 125.003 175.098 125.982 175.286 126.929C175.393 127.47 175.04 127.996 174.499 128.104C173.957 128.211 173.431 127.858 173.324 127.317C173.111 126.243 173 125.134 173 124V119.75C173 119.198 173.448 118.75 174 118.75ZM274 118.75C274.552 118.75 275 119.198 275 119.75V124C275 125.134 274.889 126.243 274.676 127.317C274.569 127.858 274.043 128.211 273.501 128.104C272.96 127.996 272.607 127.47 272.714 126.929C272.902 125.982 273 125.003 273 124V119.75C273 119.198 273.448 118.75 274 118.75ZM176.139 132.059C176.598 131.752 177.219 131.875 177.526 132.334C178.621 133.97 180.03 135.379 181.666 136.474C182.125 136.781 182.248 137.402 181.941 137.861C181.634 138.32 181.013 138.443 180.554 138.136C178.7 136.895 177.105 135.3 175.864 133.446C175.557 132.987 175.68 132.366 176.139 132.059ZM271.861 132.059C272.32 132.366 272.443 132.987 272.136 133.446C270.895 135.3 269.3 136.895 267.446 138.136C266.987 138.443 266.366 138.32 266.059 137.861C265.752 137.402 265.875 136.781 266.334 136.474C267.97 135.379 269.379 133.97 270.474 132.334C270.781 131.875 271.402 131.752 271.861 132.059ZM185.896 139.501C186.004 138.96 186.53 138.607 187.071 138.714C188.018 138.902 188.997 139 190 139H194.25C194.802 139 195.25 139.448 195.25 140C195.25 140.552 194.802 141 194.25 141H190C188.866 141 187.757 140.889 186.683 140.676C186.142 140.569 185.789 140.043 185.896 139.501ZM262.104 139.501C262.211 140.043 261.858 140.569 261.317 140.676C260.243 140.889 259.134 141 258 141H253.75C253.198 141 252.75 140.552 252.75 140C252.75 139.448 253.198 139 253.75 139H258C259.003 139 259.982 138.902 260.929 138.714C261.47 138.607 261.996 138.96 262.104 139.501ZM201.75 140C201.75 139.448 202.198 139 202.75 139H211.25C211.802 139 212.25 139.448 212.25 140C212.25 140.552 211.802 141 211.25 141H202.75C202.198 141 201.75 140.552 201.75 140ZM218.75 140C218.75 139.448 219.198 139 219.75 139H228.25C228.802 139 229.25 139.448 229.25 140C229.25 140.552 228.802 141 228.25 141H219.75C219.198 141 218.75 140.552 218.75 140ZM235.75 140C235.75 139.448 236.198 139 236.75 139H245.25C245.802 139 246.25 139.448 246.25 140C246.25 140.552 245.802 141 245.25 141H236.75C236.198 141 235.75 140.552 235.75 140Z" fill="#4D4D4D"/>
<path d="M0 0V-2H-2V0H0ZM325 0H327V-2H325V0ZM325 180V182H327V180H325ZM0 180H-2V182H0V180ZM0 2H325V-2H0V2ZM323 0V180H327V0H323ZM325 178H0V182H325V178ZM2 180V0H-2V180H2Z" fill="#767676"/>
</svg>
