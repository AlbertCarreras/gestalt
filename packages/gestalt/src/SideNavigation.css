.ulItem {
  list-style: none;
  padding-inline-start: 0;
}

.liItem {
  margin-bottom: 0;
  margin-top: 0;
}
