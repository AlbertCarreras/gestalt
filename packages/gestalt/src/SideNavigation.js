// @flow strict
import { cloneElement, Children, type Node } from 'react';
import classnames from 'classnames';
import Box from './Box.js';
import Flex from './Flex.js';
import Divider from './Divider.js';
import styles from './SideNavigation.css';
import borderStyles from './Borders.css';
import SideNavigationSection from './SideNavigationSection.js';
import SideNavigationItem from './SideNavigationItem.js';

type Props = {|
  /**
   * String that clients such as VoiceOver will read to describe the element.
   */
  accessibilityLabel: string,
  /**
   * The content shown in SideNavigation. See [subcomponents](https://gestalt.pinterest.systems/sidenavigation#Subcomponents).
   */
  children: Node,
  /**
   * Content to display at the bottom of SideNavigation. Open slot available to display other functionality required in the page. See the [Footer variant](https://gestalt.pinterest.systems/sidenavigation#Header) to learn more.
   */
  footer?: Node,
  /**
   * Content to display at the top of SideNavigation. Open slot used for controlling the display of navigation items. See the [Header variant](https://gestalt.pinterest.systems/sidenavigation#Header) to learn more.
   */
  header?: Node,
  /**
   * Displays a border in SideNavigation. See the [Border](https://gestalt.pinterest.systems/sidenavigation#Border) variant for more info.
   */
  showBorder?: boolean,
|};

/**
 * Use [SideNavigation](https://gestalt.pinterest.systems/sidenavigation) for side navigation on a page or section.
 * **NOTE**This component is on alpha phase, still under developoment. The component will support three levels and keyboard navigation. The component will change behavior and the API might also change in future component version releases.**NOTE**
 *
 * ![SideNavigation light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/SideNavigation.spec.mjs-snapshots/SideNavigation-chromium-darwin.png)
 * ![SideNavigation dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/SideNavigation-dark.spec.mjs-snapshots/SideNavigation-dark-chromium-darwin.png)
 *
 */
export default function SideNavigation({
  accessibilityLabel,
  children,
  footer,
  header,
  showBorder,
}: Props): Node {
  let hasFirstNavigationItem = false;

  const navigationChildren = [];

  // TO BE CLEANED UP

  Children.toArray(children).forEach((child) => {
    if (child.type.displayName === 'SideNavigation.Section') {
      if (!hasFirstNavigationItem) {
        hasFirstNavigationItem = true;
        navigationChildren.push(child);
      } else {
        navigationChildren.push(cloneElement(child, { _hasMarginTop: true }));
      }
    } else if (child.type.displayName === 'SideNavigation.Item') {
      if (!hasFirstNavigationItem) {
        hasFirstNavigationItem = true;
      }
      navigationChildren.push(child);
    } else if (!child?.type?.displayName?.startsWith('SideNavigation')) {
      Children.toArray(child.props.children).forEach((subchild) => {
        if (subchild.type.displayName === 'SideNavigation.Section') {
          if (!hasFirstNavigationItem) {
            hasFirstNavigationItem = true;
            navigationChildren.push(subchild);
          } else {
            navigationChildren.push(cloneElement(subchild, { _hasMarginTop: true }));
          }
        } else {
          navigationChildren.push(subchild);
        }
      });
    }
  });

  return (
    <div className={showBorder ? classnames(borderStyles.borderRight) : undefined}>
      <Box
        minWidth={280}
        width={280}
        padding={2}
        color="default"
        dangerouslySetInlineStyle={{ __style: { paddingBottom: 24 } }}
      >
        <Flex direction="column" gap={4}>
          {header ? (
            <Flex direction="column" gap={4}>
              <Box paddingX={4}>{header}</Box>
              <Divider />
            </Flex>
          ) : null}

          <Box as="nav" aria-label={accessibilityLabel}>
            <ul className={classnames(styles.ulItem)}>{navigationChildren}</ul>
          </Box>

          {footer ? (
            <Flex direction="column" gap={4}>
              <Divider />
              <Box paddingX={4}>{footer}</Box>
            </Flex>
          ) : null}
        </Flex>
      </Box>
    </div>
  );
}

SideNavigation.Item = SideNavigationItem;
SideNavigation.Section = SideNavigationSection;
