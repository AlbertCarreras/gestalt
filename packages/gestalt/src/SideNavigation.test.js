// @flow strict
import { create } from 'react-test-renderer';
import SideNavigation from './SideNavigation.js';
import Box from './Box.js';

describe('SideNavigation', () => {
  it('renders basic Item', () => {
    const tree = create(
      <SideNavigation accessibilityLabel="label">
        <SideNavigation.Item onSelect={() => {}} href="#" item={{ label: 'test', value: 'test' }} />
      </SideNavigation>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Sections', () => {
    const tree = create(
      <SideNavigation accessibilityLabel="label">
        <SideNavigation.Section label="section">
          <SideNavigation.Item
            onSelect={() => {}}
            href="#"
            item={{ label: 'test', value: 'test' }}
          />
        </SideNavigation.Section>
      </SideNavigation>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Icon + Badge/Notification + Counter + Border', () => {
    const tree = create(
      <SideNavigation showBorder accessibilityLabel="label">
        <SideNavigation.Section label="section">
          <SideNavigation.Item
            onSelect={() => {}}
            href="#"
            icon="bell"
            item={{ label: 'test', value: 'test' }}
            badge={{ text: 'New', type: 'info' }}
            counter={{ number: '20', accessibilityLabel: 'You have 20 notifications' }}
          />
          <SideNavigation.Item
            onSelect={() => {}}
            href="#"
            icon="bell"
            item={{ label: 'test', value: 'test' }}
            notificationAccessibilityLabel="You have new messages"
            counter={{ number: '20', accessibilityLabel: 'You have 20 notifications' }}
          />
        </SideNavigation.Section>
      </SideNavigation>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Header + Footer', () => {
    const tree = create(
      <SideNavigation
        accessibilityLabel="label"
        header={<Box height={100} width="100%" color="default" />}
        footer={<Box height={100} width="100%" color="default" />}
      >
        <SideNavigation.Item onSelect={() => {}} href="#" item={{ label: 'test', value: 'test' }} />
      </SideNavigation>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
