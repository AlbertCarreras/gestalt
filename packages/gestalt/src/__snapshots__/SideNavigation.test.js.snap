// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavigation renders Header + Footer 1`] = `
<div>
  <div
    className="box default paddingX2 paddingY2"
    style={
      Object {
        "minWidth": 280,
        "paddingBottom": 24,
        "width": 280,
      }
    }
  >
    <div
      className="Flex columnGap4 xsDirectionColumn"
    >
      <div
        className="FlexItem"
      >
        <div
          className="Flex columnGap4 xsDirectionColumn"
        >
          <div
            className="FlexItem"
          >
            <div
              className="box paddingX4"
            >
              <div
                className="box default"
                style={
                  Object {
                    "height": 100,
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
          <div
            className="FlexItem"
          >
            <hr
              className="divider"
            />
          </div>
        </div>
      </div>
      <div
        className="FlexItem"
      >
        <nav
          aria-label="label"
          className="box"
        >
          <ul
            className="ulItem"
          >
            <li
              className="liItem"
            >
              <a
                className="link noUnderline hideOutline tapTransition rounding2 accessibilityOutline block fullWidth pointer"
                href="#"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                rel=""
                tabIndex={0}
                target={null}
              >
                <div
                  className="box itemsCenter paddingX4 paddingY2 rounding2 xsDisplayFlex"
                  style={
                    Object {
                      "minHeight": 44,
                    }
                  }
                >
                  <div
                    className="Flex rowGap2 xsDirectionRow"
                    style={
                      Object {
                        "height": "100%",
                        "width": "100%",
                      }
                    }
                  >
                    <div
                      className="FlexItem flexGrow selfCenter"
                    >
                      <span
                        className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                      >
                        test
                      </span>
                    </div>
                  </div>
                </div>
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        className="FlexItem"
      >
        <div
          className="Flex columnGap4 xsDirectionColumn"
        >
          <div
            className="FlexItem"
          >
            <hr
              className="divider"
            />
          </div>
          <div
            className="FlexItem"
          >
            <div
              className="box paddingX4"
            >
              <div
                className="box default"
                style={
                  Object {
                    "height": 100,
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SideNavigation renders Icon + Badge/Notification + Counter + Border 1`] = `
<div
  className="borderRight"
>
  <div
    className="box default paddingX2 paddingY2"
    style={
      Object {
        "minWidth": 280,
        "paddingBottom": 24,
        "width": 280,
      }
    }
  >
    <div
      className="Flex columnGap4 xsDirectionColumn"
    >
      <div
        className="FlexItem"
      >
        <nav
          aria-label="label"
          className="box"
        >
          <ul
            className="ulItem"
          >
            <li
              className="liItem"
            >
              <div
                className="box marginBottom2 paddingX4 xsDisplayFlex"
                role="presentation"
              >
                <div
                  className="Text fontSize300 defaultText alignStart breakWord fontWeightSemiBold lineClamp"
                  style={
                    Object {
                      "WebkitLineClamp": 2,
                    }
                  }
                  title="section"
                >
                  section
                </div>
              </div>
              <ul
                className="ulItem"
              >
                <li
                  className="liItem"
                >
                  <a
                    className="link noUnderline hideOutline tapTransition rounding2 accessibilityOutline block fullWidth pointer"
                    href="#"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyPress={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchCancel={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    rel=""
                    tabIndex={0}
                    target={null}
                  >
                    <div
                      className="box itemsCenter paddingX4 paddingY2 rounding2 xsDisplayFlex"
                      style={
                        Object {
                          "minHeight": 44,
                        }
                      }
                    >
                      <div
                        className="Flex rowGap2 xsDirectionRow"
                        style={
                          Object {
                            "height": "100%",
                            "width": "100%",
                          }
                        }
                      >
                        <div
                          className="FlexItem selfCenter"
                        >
                          <div
                            aria-hidden={true}
                            className="box"
                          >
                            <svg
                              aria-hidden={true}
                              aria-label=""
                              className="icon defaultIcon"
                              height={16}
                              role="img"
                              viewBox="0 0 24 24"
                              width={16}
                            >
                              <path
                                d="test-file-stub"
                              />
                            </svg>
                          </div>
                        </div>
                        <div
                          className="FlexItem flexGrow selfCenter"
                        >
                          <span
                            className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                          >
                            test
                            <div
                              className="box marginStart1 xsDisplayInlineBlock"
                              style={
                                Object {
                                  "height": "100%",
                                }
                              }
                            >
                              <div
                                className="box xsDisplayVisuallyHidden"
                              >
                                , 
                              </div>
                              <div
                                className="Badge middle infoBase"
                              >
                                New
                              </div>
                            </div>
                          </span>
                        </div>
                        <div
                          className="FlexItem flexNone selfCenter"
                        >
                          <div
                            className="box xsDisplayVisuallyHidden"
                          >
                            , 
                          </div>
                          <div
                            aria-label="You have 20 notifications"
                            className="box marginEndN2"
                            role="status"
                          >
                            <div
                              className="Text fontSize300 defaultText alignEnd breakWord fontWeightNormal"
                            >
                              20
                               
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </a>
                </li>
                <li
                  className="liItem"
                >
                  <a
                    className="link noUnderline hideOutline tapTransition rounding2 accessibilityOutline block fullWidth pointer"
                    href="#"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyPress={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchCancel={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    rel=""
                    tabIndex={0}
                    target={null}
                  >
                    <div
                      className="box itemsCenter paddingX4 paddingY2 rounding2 xsDisplayFlex"
                      style={
                        Object {
                          "minHeight": 44,
                        }
                      }
                    >
                      <div
                        className="Flex rowGap2 xsDirectionRow"
                        style={
                          Object {
                            "height": "100%",
                            "width": "100%",
                          }
                        }
                      >
                        <div
                          className="FlexItem selfCenter"
                        >
                          <div
                            aria-hidden={true}
                            className="box"
                          >
                            <svg
                              aria-hidden={true}
                              aria-label=""
                              className="icon defaultIcon"
                              height={16}
                              role="img"
                              viewBox="0 0 24 24"
                              width={16}
                            >
                              <path
                                d="test-file-stub"
                              />
                            </svg>
                          </div>
                        </div>
                        <div
                          className="FlexItem flexGrow selfCenter"
                        >
                          <span
                            className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                          >
                            test
                            <div
                              className="box marginStart1 xsDisplayInlineBlock"
                              style={
                                Object {
                                  "height": "100%",
                                }
                              }
                            >
                              <div
                                className="box xsDisplayVisuallyHidden"
                              >
                                , 
                              </div>
                              <div
                                aria-label="You have new messages"
                                className="box circle primary"
                                role="status"
                                style={
                                  Object {
                                    "height": 8,
                                    "width": 8,
                                  }
                                }
                              />
                            </div>
                          </span>
                        </div>
                        <div
                          className="FlexItem flexNone selfCenter"
                        >
                          <div
                            className="box xsDisplayVisuallyHidden"
                          >
                            , 
                          </div>
                          <div
                            aria-label="You have 20 notifications"
                            className="box marginEndN2"
                            role="status"
                          >
                            <div
                              className="Text fontSize300 defaultText alignEnd breakWord fontWeightNormal"
                            >
                              20
                               
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </a>
                </li>
              </ul>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</div>
`;

exports[`SideNavigation renders Sections 1`] = `
<div>
  <div
    className="box default paddingX2 paddingY2"
    style={
      Object {
        "minWidth": 280,
        "paddingBottom": 24,
        "width": 280,
      }
    }
  >
    <div
      className="Flex columnGap4 xsDirectionColumn"
    >
      <div
        className="FlexItem"
      >
        <nav
          aria-label="label"
          className="box"
        >
          <ul
            className="ulItem"
          >
            <li
              className="liItem"
            >
              <div
                className="box marginBottom2 paddingX4 xsDisplayFlex"
                role="presentation"
              >
                <div
                  className="Text fontSize300 defaultText alignStart breakWord fontWeightSemiBold lineClamp"
                  style={
                    Object {
                      "WebkitLineClamp": 2,
                    }
                  }
                  title="section"
                >
                  section
                </div>
              </div>
              <ul
                className="ulItem"
              >
                <li
                  className="liItem"
                >
                  <a
                    className="link noUnderline hideOutline tapTransition rounding2 accessibilityOutline block fullWidth pointer"
                    href="#"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyPress={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchCancel={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    rel=""
                    tabIndex={0}
                    target={null}
                  >
                    <div
                      className="box itemsCenter paddingX4 paddingY2 rounding2 xsDisplayFlex"
                      style={
                        Object {
                          "minHeight": 44,
                        }
                      }
                    >
                      <div
                        className="Flex rowGap2 xsDirectionRow"
                        style={
                          Object {
                            "height": "100%",
                            "width": "100%",
                          }
                        }
                      >
                        <div
                          className="FlexItem flexGrow selfCenter"
                        >
                          <span
                            className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                          >
                            test
                          </span>
                        </div>
                      </div>
                    </div>
                  </a>
                </li>
              </ul>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</div>
`;

exports[`SideNavigation renders basic Item 1`] = `
<div>
  <div
    className="box default paddingX2 paddingY2"
    style={
      Object {
        "minWidth": 280,
        "paddingBottom": 24,
        "width": 280,
      }
    }
  >
    <div
      className="Flex columnGap4 xsDirectionColumn"
    >
      <div
        className="FlexItem"
      >
        <nav
          aria-label="label"
          className="box"
        >
          <ul
            className="ulItem"
          >
            <li
              className="liItem"
            >
              <a
                className="link noUnderline hideOutline tapTransition rounding2 accessibilityOutline block fullWidth pointer"
                href="#"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                rel=""
                tabIndex={0}
                target={null}
              >
                <div
                  className="box itemsCenter paddingX4 paddingY2 rounding2 xsDisplayFlex"
                  style={
                    Object {
                      "minHeight": 44,
                    }
                  }
                >
                  <div
                    className="Flex rowGap2 xsDirectionRow"
                    style={
                      Object {
                        "height": "100%",
                        "width": "100%",
                      }
                    }
                  >
                    <div
                      className="FlexItem flexGrow selfCenter"
                    >
                      <span
                        className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
                      >
                        test
                      </span>
                    </div>
                  </div>
                </div>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
